app/scripts/metamask-controller.js

```diff
@@ -238,7 +237,7 @@ export default class MetamaskController extends EventEmitter {
     });
     this.tokenListController = new TokenListController({
       chainId: hexToDecimal(this.networkController.getCurrentChainId()),
-      useStaticTokenList: this.preferencesController.store.getState()
+      useStaticTokenList: !this.preferencesController.store.getState()
```

Comment:
And then just not invert this value? Why would we do it this way?

Reply:
The `Use Token Detection` toggle in extension in ON by default to use the token service api, so we need the value of `useStaticTokenList` to be true. while in the controller, the `true` value for `useStaticTokenList` means we are using the static token list.